src/gt4py/next/program_processors/runners/dace_iterator/itir_to_sdfg.py

```diff
@@ -0,0 +1,355 @@
+        assert isinstance(node.output, itir.SymRef)
+
+        neighbor_tables = filter_neighbor_tables(self.offset_provider)
+        input_names = [str(inp.id) for inp in node.inputs]
+        conn_names = [connectivity_identifier(offset) for offset, _ in neighbor_tables]
+        output_names = [str(node.output.id)]
+
+        # Create the closure's nested SDFG and single state.
+        closure_sdfg = dace.SDFG(name="closure")
+        closure_state = closure_sdfg.add_state("closure_entry")
+
+        # Add DaCe arrays for inputs, output and connectivities to closure SDFG.
+        for name in [*input_names, *conn_names, *output_names]:
+            assert name not in closure_sdfg.arrays or (name in input_names and name in output_names)
+            if name not in closure_sdfg.arrays:
+                closure_sdfg.add_array(
+                    name,
+                    shape=array_table[name].shape,
+                    strides=array_table[name].strides,
+                    dtype=array_table[name].dtype,
+                )
+
+        # Get output domain of the closure
+        program_arg_syms: dict[str, ValueExpr | IteratorExpr] = {}
+        for name, type_ in self.storages.items():
+            if isinstance(type_, ts.ScalarType):
+                dtype = as_dace_type(type_)
+                closure_sdfg.add_symbol(name, dtype)
+                out_name = unique_var_name()
+                closure_sdfg.add_scalar(out_name, dtype, transient=True)
+                out_tasklet = closure_state.add_tasklet(
+                    f"get_{name}", {}, {"__result"}, f"__result = {name}"
+                )
+                access = closure_state.add_access(out_name)
+                value = ValueExpr(access, dtype)
+                memlet = create_memlet_at(out_name, ("0",))
+                closure_state.add_edge(out_tasklet, "__result", access, None, memlet)
+                program_arg_syms[name] = value
+        domain_ctx = Context(closure_sdfg, closure_state, program_arg_syms)
+        closure_domain = self._visit_domain(node.domain, domain_ctx)
+        map_domain = {
+            f"i_{dim}": f"{lb.value.data}:{ub.value.data}" for dim, (lb, ub) in closure_domain
+        }
+
+        # Create an SDFG for the tasklet that computes a single item of the output domain.
+        index_domain = {dim: f"i_{dim}" for dim, _ in closure_domain}
+
+        input_arrays = [
+            (closure_sdfg.arrays[name], name, self.storages[name]) for name in input_names
+        ]
+        conn_arrays = [(closure_sdfg.arrays[name], name) for name in conn_names]
+
+        context, results = closure_to_tasklet_sdfg(
+            node,
+            self.offset_provider,
+            index_domain,
+            input_arrays,
+            conn_arrays,
+            self.node_types,
+        )
+
+        # Map SDFG tasklet arguments to parameters
+        input_memlets = [
+            create_memlet_full(name, closure_sdfg.arrays[name]) for name in input_names
```

Comment:
Suppose that the same container is used as both input and output. In this case, the entry and exit memlet should use the same subset. So we cannot mix memlet_full and memlet_at for in/out data.

Reply:
In fact, the nested "tasklet SDFG" will refer to the same data with different names in that case which is undefined. (it seems simplify resolves this well, but we shouldn't rely on it.)

I propose to put a temporary output a transient in the output and later copy back to the original array. Can someone confirm that this is in line with itir semantics?    Namely this is cases where the same field is used as input and output to closures? (@petiaccja ?)

Reply:
I pushed a version with the copy.

Reply:
Summary based on our chat:
- In ITIR semantics, it's only allowed as long as there are no shifts in the stencil
- Temporaries in the SDFG are the most sensible (if not the only) way
- DaCe should be able to optimize out the temporaries
So I guess it's fine.